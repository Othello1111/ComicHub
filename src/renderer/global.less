@import '~antd/lib/style/themes/default.less';

body {
  text-rendering: optimizeLegibility;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  color: #000;
  line-height: 1.5;
  background-color: #f2f4f5;
}
