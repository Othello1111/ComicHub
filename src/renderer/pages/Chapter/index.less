@import '~antd/lib/style/themes/default.less';

.submit {
    padding-bottom: @form-item-margin-bottom;
}

