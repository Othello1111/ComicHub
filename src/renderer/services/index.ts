import request from '../utils/request';

interface PostData {
    url?: string;

    [index: string]: string;
}

export async function getMenuList() {
    return request('/v1/menu');
}

export async function postItem(params: PostData = {}) {
    const { url, ...rest } = params;
    if (!url) {
        return;
    }
    const realUrl = `/v1/${url}`;
    return request(realUrl, {
        method: 'POST',
        body: rest,
    });
}
