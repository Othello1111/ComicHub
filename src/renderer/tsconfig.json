{
    "compilerOptions": {
        "target": "esnext",
        "module": "esnext",
        "moduleResolution": "node",
        "importHelpers": true,
        "jsx": "preserve",
        "esModuleInterop": true,
        "sourceMap": true,
        "strict": true,
        "emitDecoratorMetadata": true,
        "experimentalDecorators": true,
        "baseUrl": ".",
        "paths": {
            "@/*": ["./*"]
        },
        "allowSyntheticDefaultImports": true
    }
}
