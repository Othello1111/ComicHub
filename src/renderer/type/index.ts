export interface SharedState {
    currentUrl: string;
    currentType: string;
}
export interface MenuItem {
    value: string;
    enabled: boolean;
    base: string;
    name: string;
}
export interface IFormData {
    url: string;
    name: string;
}

export interface IOptionData {
    value: string | number;
    name: string;
    enabled: boolean;
}
