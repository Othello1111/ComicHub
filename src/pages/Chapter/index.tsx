import { Avatar, Button, message } from 'antd';
import { Location } from 'history'
import { getLanguageData } from '../../locales';
import React, { Fragment, useEffect, useState } from 'react';
import { renderDate, getQuery, history } from '../../utils';
import styles from './index.less';
import DumpTable from '../../components/DumpTable';
import { TypeConfig } from '../../type';
import { ISearchItem } from '../../../server/type';
import { postItem } from '../../services';

const searchColumns = [
  {
    dataIndex: 'id',
    title: 'ID',
  },
  {
    dataIndex: 'title',
    title: getLanguageData('page.Chapter.table.title'),
  },
  {
    dataIndex: 'author',
    title: getLanguageData('page.Chapter.table.author'),
  },
  {
    dataIndex: 'url',
    title: getLanguageData('page.Chapter.table.url'),
    render: (text: string) => (
      <a title={text} target="_blank" href={text} rel="noopener noreferrer">
        {text}
      </a>
    ),
  },

  {
    dataIndex: 'area',
    title: getLanguageData('page.Chapter.table.area'),
  },
  {
    dataIndex: 'category',
    title: getLanguageData('page.Chapter.table.category'),
  },
  {
    dataIndex: 'cover',
    title: getLanguageData('page.Chapter.table.cover'),
    render: (text: string) =>
      text ? (
        <a target="_blank" href={text} rel="noopener noreferrer">
          <Avatar src={text}/>
        </a>
      ) : (
        ''
      ),
  },
  {
    dataIndex: 'create_time',
    title: getLanguageData('page.Chapter.table.create_time'),
    render: renderDate,
  },
];

interface Props {
  location: Location;
}

const Chapter: React.FunctionComponent<Props> = (props: Props) => {
  const {
    location,
  } = props;
  const [selectedRows, setSelectedRows] = useState<ISearchItem[]>([]);
  const checkType = 'radio';
  const [list, setList] = useState<ISearchItem[]>([]);
  const query = getQuery(location.search);
  useEffect(() => {
    postItem({
      url: query.url,
      name: query.name,
      type: TypeConfig.search,
    }).then(data => {
      setList(data);
    });
  }, []);

  function handleSelectRows(value: ISearchItem[]): void {
    setSelectedRows(value);
  }

  function handleChapterSubmit(): void {
    if (!selectedRows || selectedRows.length === 0) {
      message.error(getLanguageData('page.Chapter.select.tip'));
      return;
    }
    const [item] = selectedRows;
    const link = `/${TypeConfig.download}?url=${query.url}&name=${encodeURIComponent(item.url)}`;
    history.push(link);
  }

  return (
    <Fragment>
      <div className={styles.submit}>
        <Button
          type="primary"
          onClick={handleChapterSubmit}
          disabled={selectedRows.length === 0}
        >
          {getLanguageData('component.button.submit')}
        </Button>
      </div>
      <DumpTable
        checkType={checkType}
        selectedRows={selectedRows}
        data={list}
        columns={searchColumns}
        onSelectRow={handleSelectRows}
      />
    </Fragment>
  );
};


export default Chapter;
