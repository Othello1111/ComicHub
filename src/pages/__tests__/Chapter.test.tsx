import React from 'react';
import renderer, { ReactTestRenderer } from 'react-test-renderer';
import Chapter from '../Chapter';
import Store from '../../store';

it('Page: Chpter  Snapshots', () => {
  const location = { pathname: 'test', search: '?', hash: '', state: '' };

  let tree: ReactTestRenderer;
  renderer.act(() => {
    tree = renderer
      .create(<Store.Provider><Chapter location={location}/></Store.Provider>);
  });
  // @ts-ignore
  expect(tree.toJSON()).toMatchSnapshot();
});
