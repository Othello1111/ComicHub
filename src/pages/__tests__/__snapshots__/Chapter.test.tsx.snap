// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page: Chpter  Snapshots 1`] = `
Array [
  " ",
  <div
    className="submit-button"
  >
    <button
      className="ant-btn ant-btn-primary"
      disabled={true}
      onClick={[Function]}
      type="button"
    >
      <span>
        提 交
      </span>
    </button>
  </div>,
  <div
    className="table"
  >
    <div
      className="table-alert"
    >
      <div
        className="ant-alert ant-alert-info"
        data-show={true}
      >
        <i
          aria-label="icon: info-circle"
          className="anticon anticon-info-circle ant-alert-icon"
        >
          <svg
            aria-hidden="true"
            className=""
            data-icon="info-circle"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm32 664c0 4.4-3.6 8-8 8h-48c-4.4 0-8-3.6-8-8V456c0-4.4 3.6-8 8-8h48c4.4 0 8 3.6 8 8v272zm-32-344a48.01 48.01 0 0 1 0-96 48.01 48.01 0 0 1 0 96z"
            />
          </svg>
        </i>
        <span
          className="ant-alert-message"
        >
          已选择
           
          <a
            style={
              Object {
                "fontWeight": 600,
              }
            }
          >
            0
          </a>
           
          项
           
          <a
            onClick={[Function]}
            style={
              Object {
                "marginLeft": 24,
              }
            }
          >
            清空
          </a>
        </span>
        <span
          className="ant-alert-description"
        />
      </div>
    </div>
    <div
      className="ant-table-wrapper"
    >
      <div
        className="ant-spin-nested-loading"
      >
        <div
          className="ant-spin-container"
        >
          <div
            className="ant-table ant-table-small ant-table-bordered ant-table-empty ant-table-scroll-position-left"
            style={Object {}}
          >
            <div
              className="ant-table-content"
            >
              <div
                className="ant-table-body"
                onScroll={[Function]}
                onWheel={[Function]}
                style={Object {}}
              >
                <table
                  className=""
                  style={Object {}}
                >
                  <colgroup>
                    <col
                      className="ant-table-selection-col"
                      style={
                        Object {
                          "minWidth": undefined,
                          "width": undefined,
                        }
                      }
                    />
                    <col
                      style={
                        Object {
                          "minWidth": undefined,
                          "width": undefined,
                        }
                      }
                    />
                    <col
                      style={
                        Object {
                          "minWidth": undefined,
                          "width": undefined,
                        }
                      }
                    />
                    <col
                      style={
                        Object {
                          "minWidth": undefined,
                          "width": undefined,
                        }
                      }
                    />
                    <col
                      style={
                        Object {
                          "minWidth": undefined,
                          "width": undefined,
                        }
                      }
                    />
                    <col
                      style={
                        Object {
                          "minWidth": undefined,
                          "width": undefined,
                        }
                      }
                    />
                    <col
                      style={
                        Object {
                          "minWidth": undefined,
                          "width": undefined,
                        }
                      }
                    />
                    <col
                      style={
                        Object {
                          "minWidth": undefined,
                          "width": undefined,
                        }
                      }
                    />
                    <col
                      style={
                        Object {
                          "minWidth": undefined,
                          "width": undefined,
                        }
                      }
                    />
                  </colgroup>
                  <thead
                    className="ant-table-thead"
                  >
                    <tr
                      style={
                        Object {
                          "height": null,
                        }
                      }
                    >
                      <th
                        className="ant-table-selection-column"
                      >
                        <span
                          className="ant-table-header-column"
                        >
                          <div>
                            <span
                              className="ant-table-column-title"
                            />
                            <span
                              className="ant-table-column-sorter"
                            />
                          </div>
                        </span>
                      </th>
                      <th
                        className=""
                      >
                        <span
                          className="ant-table-header-column"
                        >
                          <div>
                            <span
                              className="ant-table-column-title"
                            >
                              ID
                            </span>
                            <span
                              className="ant-table-column-sorter"
                            />
                          </div>
                        </span>
                      </th>
                      <th
                        className=""
                      >
                        <span
                          className="ant-table-header-column"
                        >
                          <div>
                            <span
                              className="ant-table-column-title"
                            >
                              名称
                            </span>
                            <span
                              className="ant-table-column-sorter"
                            />
                          </div>
                        </span>
                      </th>
                      <th
                        className=""
                      >
                        <span
                          className="ant-table-header-column"
                        >
                          <div>
                            <span
                              className="ant-table-column-title"
                            >
                              作者
                            </span>
                            <span
                              className="ant-table-column-sorter"
                            />
                          </div>
                        </span>
                      </th>
                      <th
                        className=""
                      >
                        <span
                          className="ant-table-header-column"
                        >
                          <div>
                            <span
                              className="ant-table-column-title"
                            >
                              链接
                            </span>
                            <span
                              className="ant-table-column-sorter"
                            />
                          </div>
                        </span>
                      </th>
                      <th
                        className=""
                      >
                        <span
                          className="ant-table-header-column"
                        >
                          <div>
                            <span
                              className="ant-table-column-title"
                            >
                              地区
                            </span>
                            <span
                              className="ant-table-column-sorter"
                            />
                          </div>
                        </span>
                      </th>
                      <th
                        className=""
                      >
                        <span
                          className="ant-table-header-column"
                        >
                          <div>
                            <span
                              className="ant-table-column-title"
                            >
                              分类
                            </span>
                            <span
                              className="ant-table-column-sorter"
                            />
                          </div>
                        </span>
                      </th>
                      <th
                        className=""
                      >
                        <span
                          className="ant-table-header-column"
                        >
                          <div>
                            <span
                              className="ant-table-column-title"
                            >
                              封面
                            </span>
                            <span
                              className="ant-table-column-sorter"
                            />
                          </div>
                        </span>
                      </th>
                      <th
                        className=""
                      >
                        <span
                          className="ant-table-header-column"
                        >
                          <div>
                            <span
                              className="ant-table-column-title"
                            >
                              爬取时间
                            </span>
                            <span
                              className="ant-table-column-sorter"
                            />
                          </div>
                        </span>
                      </th>
                    </tr>
                  </thead>
                  <tbody
                    className="ant-table-tbody"
                  />
                </table>
              </div>
              <div
                className="ant-table-placeholder"
              >
                <div
                  className="ant-empty ant-empty-normal"
                >
                  <div
                    className="ant-empty-image"
                  >
                    <svg
                      height="41"
                      viewBox="0 0 64 41"
                      width="64"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <g
                        fill="none"
                        fillRule="evenodd"
                        transform="translate(0 1)"
                      >
                        <ellipse
                          cx="32"
                          cy="33"
                          fill="#F5F5F5"
                          rx="32"
                          ry="7"
                        />
                        <g
                          fillRule="nonzero"
                          stroke="#D9D9D9"
                        >
                          <path
                            d="M55 12.76L44.854 1.258C44.367.474 43.656 0 42.907 0H21.093c-.749 0-1.46.474-1.947 1.257L9 12.761V22h46v-9.24z"
                          />
                          <path
                            d="M41.613 15.931c0-1.605.994-2.93 2.227-2.931H55v18.137C55 33.26 53.68 35 52.05 35h-40.1C10.32 35 9 33.259 9 31.137V13h11.16c1.233 0 2.227 1.323 2.227 2.928v.022c0 1.605 1.005 2.901 2.237 2.901h14.752c1.232 0 2.237-1.308 2.237-2.913v-.007z"
                            fill="#FAFAFA"
                          />
                        </g>
                      </g>
                    </svg>
                  </div>
                  <p
                    className="ant-empty-description"
                  >
                    No Data
                  </p>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  " ",
]
`;
